kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ClusterService.java

```diff
@@ -273,4 +284,47 @@ private KafkaCluster updateCluster(String topicToDelete, String clusterName,
   }
 
 
+  public Flux<ConsumerGroupDeleteResult> deleteConsumerGroups(String clusterName,
+                                                              Mono<ConsumerGroupIds> groupIds) {
+    Optional<Flux<ConsumerGroupDeleteResult>> result =
+        clustersStorage.getClusterByName(clusterName)
+            .map(cluster -> kafkaService.getOrCreateAdminClient(cluster)
+                .map(ExtendedAdminClient::getAdminClient)
+                .flatMap(kafkaClient -> groupIds
+                    .map(ConsumerGroupIds::getIds)
+                    .map(kafkaClient::deleteConsumerGroups)
+                    .map(DeleteConsumerGroupsResult::deletedGroups)
+                    .map(this::getConsumerGroupDeletesResult))
+                .flatMapMany(Flux::fromIterable));
+    return result.orElse(Flux.empty());
+  }
+
+  @NotNull
+  private List<ConsumerGroupDeleteResult> getConsumerGroupDeletesResult(
+      Map<String, KafkaFuture<Void>> futuresMap) {
+    return futuresMap.entrySet()
+        .stream()
+        .map(this::getConsumerGroupDeleteResult)
+        .collect(Collectors.toList());
+  }
+
+  @NotNull
+  @SneakyThrows
+  private ConsumerGroupDeleteResult getConsumerGroupDeleteResult(
+      Map.Entry<String, KafkaFuture<Void>> groupIdAndFuture) {
+    try {
+      KafkaFuture<Void> future = groupIdAndFuture.getValue();
+      Void result = future.get(1, TimeUnit.SECONDS);
```

Comment:
Please remove blocking operation -- use ClusterUtils#toMono

Reply:
Yes, thank you. Fixed!